File: services/java-spring/test_service/services/echo/EchoService.java
package test_service.services.echo;

import java.math.BigDecimal;
import java.time.*;
import java.util.*;

import test_service.models.*;

public interface EchoService {
	String echoBodyString(String body);
	Message echoBody(Message body);
	Parameters echoQuery(int intQuery, long longQuery, float floatQuery, double doubleQuery, BigDecimal decimalQuery, boolean boolQuery, String stringQuery, String stringOptQuery, String stringDefaultedQuery, String[] stringArrayQuery, UUID uuidQuery, LocalDate dateQuery, LocalDate[] dateArrayQuery, LocalDateTime datetimeQuery, Choice enumQuery);
	Parameters echoHeader(int intHeader, long longHeader, float floatHeader, double doubleHeader, BigDecimal decimalHeader, boolean boolHeader, String stringHeader, String stringOptHeader, String stringDefaultedHeader, String[] stringArrayHeader, UUID uuidHeader, LocalDate dateHeader, LocalDate[] dateArrayHeader, LocalDateTime datetimeHeader, Choice enumHeader);
	UrlParameters echoUrlParams(int intUrl, long longUrl, float floatUrl, double doubleUrl, BigDecimal decimalUrl, boolean boolUrl, String stringUrl, UUID uuidUrl, LocalDate dateUrl, LocalDateTime datetimeUrl, Choice enumUrl);
	EchoEverythingResponse echoEverything(Message body, float floatQuery, boolean boolQuery, UUID uuidHeader, LocalDateTime datetimeHeader, LocalDate dateUrl, BigDecimal decimalUrl);
	SameOperationNameResponse sameOperationName();
}

File: services/scala-play/services/echo/EchoService.scala
package services.echo

import com.google.inject.ImplementedBy
import scala.concurrent.Future
import models._

@ImplementedBy(classOf[services.EchoService])
trait IEchoService {
  def echoBodyString(body: String): Future[EchoBodyStringResponse]
  def echoBody(body: Message): Future[EchoBodyResponse]
  def echoQuery(intQuery: Int, longQuery: Long, floatQuery: Float, doubleQuery: Double, decimalQuery: BigDecimal, boolQuery: Boolean, stringQuery: String, stringOptQuery: Option[String], stringDefaultedQuery: String, stringArrayQuery: List[String], uuidQuery: java.util.UUID, dateQuery: java.time.LocalDate, dateArrayQuery: List[java.time.LocalDate], datetimeQuery: java.time.LocalDateTime, enumQuery: Choice): Future[EchoQueryResponse]
  def echoHeader(intHeader: Int, longHeader: Long, floatHeader: Float, doubleHeader: Double, decimalHeader: BigDecimal, boolHeader: Boolean, stringHeader: String, stringOptHeader: Option[String], stringDefaultedHeader: String, stringArrayHeader: List[String], uuidHeader: java.util.UUID, dateHeader: java.time.LocalDate, dateArrayHeader: List[java.time.LocalDate], datetimeHeader: java.time.LocalDateTime, enumHeader: Choice): Future[EchoHeaderResponse]
  def echoUrlParams(intUrl: Int, longUrl: Long, floatUrl: Float, doubleUrl: Double, decimalUrl: BigDecimal, boolUrl: Boolean, stringUrl: String, uuidUrl: java.util.UUID, dateUrl: java.time.LocalDate, datetimeUrl: java.time.LocalDateTime, enumUrl: Choice): Future[EchoUrlParamsResponse]
  def echoEverything(uuidHeader: java.util.UUID, datetimeHeader: java.time.LocalDateTime, body: Message, dateUrl: java.time.LocalDate, decimalUrl: BigDecimal, floatQuery: Float, boolQuery: Boolean): Future[EchoEverythingResponse]
  def sameOperationName(): Future[SameOperationNameResponse]
}

sealed trait EchoBodyStringResponse
object EchoBodyStringResponse {
  case class Ok(body: String) extends EchoBodyStringResponse
}

sealed trait EchoBodyResponse
object EchoBodyResponse {
  case class Ok(body: Message) extends EchoBodyResponse
}

sealed trait EchoQueryResponse
object EchoQueryResponse {
  case class Ok(body: Parameters) extends EchoQueryResponse
}

sealed trait EchoHeaderResponse
object EchoHeaderResponse {
  case class Ok(body: Parameters) extends EchoHeaderResponse
}

sealed trait EchoUrlParamsResponse
object EchoUrlParamsResponse {
  case class Ok(body: UrlParameters) extends EchoUrlParamsResponse
}

sealed trait EchoEverythingResponse
object EchoEverythingResponse {
  case class Ok(body: Everything) extends EchoEverythingResponse
  case class Forbidden() extends EchoEverythingResponse
}

sealed trait SameOperationNameResponse
object SameOperationNameResponse {
  case class Ok() extends SameOperationNameResponse
  case class Forbidden() extends SameOperationNameResponse
}
